csharp/src/SeedLang.Common/TextPosition.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021 The Aha001 Team.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+
+namespace SeedLang.Common {
+  // An immutable class to represent a position in a plaintext source code.
+  public class TextPosition : IComparable<TextPosition>, IEquatable<TextPosition> {
+    public int Line { get; }
+    public int Column { get; }
+
+    public TextPosition(int line, int column) {
+      Line = line;
+      Column = column;
+    }
+
+    public override int GetHashCode() {
+      return Tuple.Create(Line, Column).GetHashCode();
+    }
+
+    public override string ToString() {
+      return $"Ln {Line}, Col {Column}";
+    }
+
+    public int CompareTo(TextPosition pos) {
+      if (pos is null) {
```

Comment:
If it's a struct, all the null check in this classes can be removed safely.

Reply:
I've made BlockPosition and TextPosition structs as you suggested